### ISSUE TITLE
suspicious state json-jsonlint (weird error)

### ISSUE THREAD

**Issue by ejmg** [flycheck/flycheck#1415]
Hi there, I actually have a weird bug with Flycheck wrt JSON. It is weird because it appears I have everything installed as required **and** Flycheck seems to agree with that... nevertheless, `json-jsonlint` is throwing a `suspicious state` error.

## Error

Opening any JSON File throws the following error:

```bash
Suspicious state from syntax checker json-jsonlint: Flycheck checker json-jsonlint returned non-zero exit code 1, but its output contained no errors: Error: option -c not recognized.  Use "jsonlint --help" for usage information.

Try installing a more recent version of json-jsonlint, and please open a bug report if the issue persists in the latest release.  Thanks!
```

This is triggered upon saving, editing, etc.

Disabling Flycheck mode makes the behavior go away but that is obviously undesired.

The error is *not reproducible* with `emacs -Q` but that isn't surprising. This is definitely either flycheck or my system that is causing this.

## Expected Behavior

Basic syntax checking for JSON files.

## Environment

- OS: Ubuntu, 16.04
- emacs: 25.1.1
- Flycheck version: `32snapshot (package: 20180224.1046)`
- furthermore, `flycheck-verify-setup` shows:
```bash
Syntax checkers for buffer config.json in json-mode:

  json-jsonlint
    - may enable: yes
    - executable: Found at /home/spook/.local/bin/jsonlint

  json-python-json
    - may enable: yes
    - predicate:  t
    - executable: Found at /home/spook/.pyenv/shims/python

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20180224.1046)
Emacs version:    25.1.1
System:           x86_64-unknown-linux-gnu
Window system:    x
```
This is the confusing part. I have no clue how to debug flycheck at this point because... it seems to be saying everything is okay(?).

## Further information

I don't use `npm` or javascript generally that much. Here's what I know with regards to my `npm` install and how I installed everthing else:

- npm version: `4.1.1` (up to date)
- eslint version: `v4.18.1`
- jsonlint version: `jsonlint (demjson) version 2.2.4 (2015-12-22)`

Any help is appreciated!

**Comment by ejmg** [comment]
**Well...**

In the process of writing out this issue and doing my research... I figured out the issue! Notice how `jsonlint --version` returned `jsonlint (demjson) version 2.2.4 (2015-12-22)` above? That's not jsonlint as installed by `npm`, but rather a python package `demjson` that had shadowed the executable name with its own!

I'm a big user of python virtualenvs, but it appears I accidentally installed (as a dependency) `demjson` as a dependency on my system version of python3. Uninstalling it there with `pip uninstall demjson` fixed my problems!

Hope my issue resolution can help someone else!
